l3kernel/l3pdf.dtx

```diff
@@ -89,6 +91,58 @@
 %   in for example page resource allocation
 % \end{function}
 %
+% \begin{function}[EXP, pTF, added = 2020-05-15]{\pdf_object_if_exist:n}
+%   \begin{syntax}
+%    \cs{pdf_object_if_exist_p:n} \Arg{object}
+%    \cs{pdf_object_if_exist:nTF} \Arg{object}
+%   \end{syntax}
+%   Tests whether an object with name \Arg{object} has been defined.
+% \end{function}
+%
+% \subsection{Indexed objects}
```

Comment:
Apart from
- `\pdf_object_new_indexed:nn {<class>} {<index>}`
- `\pdf_object_write_indexed:nnnn {<class>} {<index>} {<type>} {<content>}`
- `\pdf_object_ref_indexed:nn {<class>} {<index>}`

for indexed objects, is a new conditional which checks existence of indexed object needed? For example

```tex
\pdf_object_index_if_exist:nnTF {<class>} {<index>} {<true code>} {<false code>}
```

Maybe even `\pdf_object_class_if_exist:nTF {<class>} {<true code>} {<false code>}`.

Reply:
I'd wondered about that but assume that given the use case it's unlikely: at least I'd want to see a use I think before adding.

Reply:
I agree, we should wait with adding anything like that. So far the only real use case for indexed objects is tagging where they wouldn't be needed and I don't expect many others (and even if there are other types, you wouldn't ask for a random index if that exists or not imo). So let's wait and see.

Reply:
Maybe not ask for a “random” index, but in general there may be the possibility that the indexes are not simply a complete run of integers starting at 1?